src/main/org/deidentifier/arx/distributed/ARXDistributedAnonymizer.java

```diff
@@ -314,34 +337,44 @@ public ARXDistributedResult anonymize(Data data,
      * @throws RollbackRequiredException 
      * @throws IOException 
      */
-    List<Future<DataHandle>> getAnonymization(List<DataHandle> partitions,
+    List<Future<DataHandle>> getAnonymization(List<ARXPartition> partitions,
                                                       ARXConfiguration config,
                                                       DistributionStrategy distributionStrategy2,
                                                       int[] transformation) throws IOException, RollbackRequiredException {
         List<Future<DataHandle>> futures = new ArrayList<>();
-        for (DataHandle partition : partitions) {
+        for (ARXPartition partition : partitions) {
+            ARXConfiguration localConfig = config.clone();
+            if (!isNullOrEmpty(partition.getSubset())) {
+                for (PrivacyCriterion privacyCriterion : localConfig.getPrivacyModels()) {
+                    localConfig.removeCriterion(privacyCriterion);
+                    //TODO: I don't know how to do this in a general way, so right now i assume 5-Map subset
```

Comment:
OK, this one is a bit tough. My suggestion is to implement specific clone-functions into the privacy models you want to support here (K-Map is a start, but why not also support D-Presence, if K-Map works). To this, I suggest the following steps:

(1) Create an interface in org.deidentifier.arx.criteria, e.g. called "ModelWithExchangeableSubset". The interface could define the following function:

cloneAndExchangeSubset(DataSubset subset);

(2) Extend KMap (and other models you want to support) to "implements ModelWithExchangeableSubset"

(3) Then implement the function for each model. To this, you can copy the existing code from the clone function, e.g. turn this:

    @Override
    public KMap clone() {
        return new KMap(getK(), getSignificanceLevel(), ((getPopulationModel() == null) ? null : getPopulationModel().clone()), getEstimator(), ((getDataSubset() == null) ? null : getDataSubset().clone()));
    }

Into this:

    @Override
    public KMap cloneAndExchangeSubset(DataSubset subset) {
        return new KMap(getK(), getSignificanceLevel(), ((getPopulationModel() == null) ? null : getPopulationModel().clone()), getEstimator(), ((getDataSubset() == null) ? null : subset));
    }

Please not that there already exists a function that is quite similar, i.e.:
	
	public PrivacyCriterion clone(DataSubset subset) {
		....
	}
	
But you _can not_ reuse this function, as it serves another purpose.

(4) In your main code where you clone the stuff you could then do something like:

for (PrivacyCriterion privacyCriterion : localConfig.getPrivacyModels()) {
	if (privacyCriterion implements ModelWithExchangeableSubset) {
		localConfig.removeCriterion(privacyCriterion);
		localConfig.addPrivacyModel(((ModelWithExchangeableSubset)privacyCriterion).cloneAndExchangeSubset(subset));
	}
}

An additional note. This here is a bit ugly:

Set<Integer> subset = convertArrayToSet(partition.getSubset());

Maybe use RowSet or Set<Integer> from the beginning to represent the subset. Check how complicated this would be.